app/src/main/java/ryey/easer/skills/operation/hotspot/HotspotHelper.java

```diff
@@ -20,21 +20,64 @@
 package ryey.easer.skills.operation.hotspot;
 
 import android.content.Context;
+import android.net.ConnectivityManager;
 import android.net.wifi.WifiConfiguration;
 import android.net.wifi.WifiManager;
+import android.os.ResultReceiver;
 
+import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import com.orhanobut.logger.Logger;
+
+
+
+import static android.content.Context.CONNECTIVITY_SERVICE;
+
 
 class HotspotHelper {
     private static HotspotHelper instance = null;
     private WifiManager wifiManager;
+    private ConnectivityManager connectivityManager;
+
+
+    private void callStartTethering(Object internalConnectivityManager) throws ReflectiveOperationException {
+        Class internalConnectivityManagerClass = Class.forName("android.net.IConnectivityManager");
+
+        ResultReceiver dummyResultReceiver = new ResultReceiver(null);
+
+        try {
+            Method startTetheringMethod = internalConnectivityManagerClass.getDeclaredMethod("startTethering",
+                    int.class,
+                    ResultReceiver.class,
+                    boolean.class);
+
+            startTetheringMethod.invoke(internalConnectivityManager,
+                    0,
+                    dummyResultReceiver,
+                    false);
+        } catch (NoSuchMethodException e) {
+            // Newer devices have "callingPkg" String argument at the end of this method.
+            Method startTetheringMethod = internalConnectivityManagerClass.getDeclaredMethod("startTethering",
+                    int.class,
+                    ResultReceiver.class,
+                    boolean.class,
+                    String.class);
+
+            startTetheringMethod.invoke(internalConnectivityManager,
+                    0,
+                    dummyResultReceiver,
+                    false,
+                    "com.matejdro.taskertethercontrol");
```

Comment:
Based on the code comment above, shouldn't this be changed to `ryey.easer`?

Reply:
Yes, you are right